app/connector/email/src/main/java/io/syndesis/connector/email/component/EMailComponent.java

```diff
@@ -0,0 +1,236 @@
+
+    /**
+     * These fields are populated using reflection by the HandlerCustomizer class.
+     *
+     * The values are resolved then the appropriate setter called, while the original
+     * key/value pairs are removed from the options map.
+     *
+     * Note:
+     * Should a property be secret then its raw value is the property placeholder and
+     * the resolving process converts it accordingly hence the importance of doing it
+     * this way rather than using the options map directly.
+     */
+    private String protocol;
+    private boolean secure;
+    private String host;
+    private int port = -1;
+    private String username;
+    private String password;
+    private String serverCertificate;
+    private boolean unseenOnly;
+
+    // Consumer properties
+    private long delay = -1;
+    private int maxResults = 5;
+
+    EMailComponent(String componentId, String componentScheme) {
+        super(componentId, componentScheme);
+    }
+
+    public String getProtocol() {
+        //
+        // Will convert, for example,  imap to imaps
+        // if the protocol has been specified as secure
+        //
+        if (isSecure() && (protocol != null && !protocol.endsWith("s"))) {
+            return protocol + "s";
+        }
+
+        return protocol;
+    }
+
+    public void setProtocol(String protocol) {
+        this.protocol = protocol;
+    }
+
+    public boolean isSecure() {
+        return secure;
+    }
+
+    public void setSecure(boolean secure) {
+        this.secure = secure;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getServerCertificate() {
+        return serverCertificate;
+    }
+
+    public void setServerCertificate(String serverCertificate) {
+        this.serverCertificate = serverCertificate;
+    }
+
+    public boolean isUnseenOnly() {
+        return unseenOnly;
+    }
+
+    public void setUnseenOnly(boolean unseenOnly) {
+        this.unseenOnly = unseenOnly;
+    }
+
+    public long getDelay() {
+        return delay;
+    }
+
+    public void setDelay(long delay) {
+        this.delay = delay;
+    }
+
+    public int getMaxResults() {
+        return maxResults;
+    }
+
+    public void setMaxResults(int maxResults) {
+        this.maxResults = maxResults;
+    }
+
+    private Map<String, Object> bundleOptions() {
+        PropertyBuilder<Object> builder = new PropertyBuilder<Object>();
```

Comment:
Might eliminate the need for `PropertyBuilder` if `Map::compute` was used:

```java
Map<String, Object> options = new HashMap<>();
options.compute(PROTOCOL, val(this::getProtocol));
options.compute(HOST, val(this::getHost));

return options;
```

Where `val` is:
```java
private static <T, U, R> BiFunction<T, U, R> val(final Supplier<R> fn) {
    return (t, u) -> fn.get();
}
```

If we want to be fancy that is :smile: